### ISSUE TITLE
Add missing unit tests.

### ISSUE THREAD

**Issue by dmfs** [dmfs/oauth2-essentials#6]

**Comment by dmfs** [comment]
@lemonboston please help with that.

**Comment by dmfs** [comment]
@lemonboston we split the work. You take care of everything in utils and http and I'll work on the rest.

**Comment by lemonboston** [comment]
Ok, I started with `http` package.

Is it okay to refactor `BasicAuthRequestDecorator` and `BearerAuthRequestDecorator` to use `decoration` package?
So instead of
`new BasicAuthRequestDecorator<T>(request, mClientId, mClientSecret);`
it would be
`new RequestHeaderDecorator<T>(request, new BasicAuthHeaderDecoration(mCleintId, mClientSecret))`
As they are only header decorations, it would confine that relevant code better, eliminate some (kind of) code duplication and make unit testing simpler.

**Comment by dmfs** [comment]
>Is it okay to refactor `BasicAuthRequestDecorator` and `BearerAuthRequestDecorator` to use `decoration` package?

It's ok for `BasicAuthRequestDecorator` but not for `BearerAuthRequestDecorator`. The latter one is meant to be used by users of this library and I think in this case it's better to make sure you can apply this easily. I think the generic decoration approach is a bit too far fetched for users of the library.

Btw. it just occurred to me that no class should ever be named like `*Decorator`.  This sounds just wrong. While you're at it, please rename `BearerAuthRequestDecorator` to `BearerAuthenticatedRequest`.

Before moving on with this we should wait for dmfs/http-client-essentials-suite#22 and #11.

**Comment by lemonboston** [comment]
Created a PR to add the already done tests, I will get back to this later probably (will do dmfs/http-client-essentials-suite#22)

One more question:
Don't [SimpleUrlEncodedValuePairs](https://github.com/dmfs/http-client-essentials-suite/blob/master/http-client-types/src/main/java/org/dmfs/httpessentials/types/SimpleUrlEncodedValuePairs.java) and [StructuredStringFragment](https://github.com/dmfs/oauth2-essentials/blob/master/src/main/java/org/dmfs/oauth2/client/utils/StructuredStringFragment.java) do the same? If they do, which one to keep? (which place, name, which class' content)

And a note:
In order to test the request classes in `org.dmfs.oauth2.client.http.requests` package meaningfully I would need to understand these oauth2 flows in more detail, then understand the modelling, types in this module, so that would take a bit longer. Let me know if I should do that (later) or leave that to you.
(By meaningful testing I mean not just write test to cover the code I see in the class without understanding it, but test what we would expect from them.)

**Comment by dmfs** [comment]
Good point about `SimpleUrlEncodedValuePairs` and `StructuredStringFragment`. Although the implementations are different, the results is basically the same. The only exception is that  for some reason `SimpleUrlEncodedValuePairs` skips the very first result (I've no idea why).

The implementation of `SimpleUrlEncodedValuePairs` looks a bit more robust and correct. `StructuredStringFragment` will fail when the parameter names contain percent encoded characters.

I suggest to fix `SimpleUrlEncodedValuePairs` (possibly also changing the name) and use it in this project afterwards. For now just keep it and open an issue so we won't forget about it. Apparently there is no need to write a unit test (at least no in this project)

I'll take care of the requests package.

**Comment by lemonboston** [comment]
Ok, thanks, so the things left for me here (will get back to them later):
- change `BasicAuthRequestDecorator` to use 'HeaderDecorated' after #11 , plus unit test
- unit test for `BearerAuthRequestDecorator`

**Comment by dmfs** [comment]
Yep. Also rename `BearerAuthRequestDecorator` to `BearerAuthenticatedRequest` as discussed (please also update the example in README.md when you do so).
